controllers/fenceagentsremediation_controller_test.go

```diff
@@ -182,18 +205,74 @@ func getFenceAgentsRemediation(nodeName, agent string, sharedparameters map[v1al
 	}
 }
 
-// buildFarPod builds a dummy pod with FAR label and namespace
-func buildFarPod() *corev1.Pod {
-	fenceAgentsPod := &corev1.Pod{}
-	fenceAgentsPod.Labels = faPodLabels
-	fenceAgentsPod.Name = "mock-fence-agents"
-	fenceAgentsPod.Namespace = defaultNamespace
+// buildPod builds a dummy pod
+func buildPod(containerName, podName, nodeName string) *corev1.Pod {
+	pod := &corev1.Pod{}
+	pod.Name = podName
+	if podName == farPodName {
+		// only when we build FAR pod then we add it's label
+		pod.Labels = faPodLabels
+	} else {
+		// testedPod should be reside on unhealthy node
+		pod.Spec.NodeName = nodeName
+	}
+	pod.Namespace = defaultNamespace
 	container := corev1.Container{
-		Name:  "foo",
+		Name:  containerName,
 		Image: "foo",
 	}
-	fenceAgentsPod.Spec.Containers = []corev1.Container{container}
-	return fenceAgentsPod
+	pod.Spec.Containers = []corev1.Container{container}
+	return pod
+}
+
+// createRunningPod builds new pod format, create it, and set it's status as running
+func createRunningPod(containerName, podName, nodeName string) *corev1.Pod {
+	pod := buildPod(containerName, podName, nodeName)
+	Expect(k8sClient.Create(context.Background(), pod)).To(Succeed())
+	pod.Status.Phase = corev1.PodRunning
+	Expect(k8sClient.Status().Update(context.Background(), pod)).To(Succeed())
+	return pod
+}
+
+// createVA creates new volume attachment and return it's object
+func createVA(vaName, unhealthyNodeName string) *storagev1.VolumeAttachment {
+	va := &storagev1.VolumeAttachment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      vaName,
+			Namespace: defaultNamespace,
+		},
+		Spec: storagev1.VolumeAttachmentSpec{
+			Attacher: "foo",
+			Source:   storagev1.VolumeAttachmentSource{},
+			NodeName: unhealthyNodeName,
+		},
+	}
+	foo := "foo"
+	va.Spec.Source.PersistentVolumeName = &foo
+	ExpectWithOffset(1, k8sClient.Create(context.Background(), va)).To(Succeed())
+	return va
+}
+
+// verifyResourceCleanup fetches all the resources that we have crated for the test
+// and if they are still exist at the end of the test, then we clean them up for next test
+func verifyResourceCleanup(va1, va2 *storagev1.VolumeAttachment, pod *corev1.Pod) {
```

Comment:
I understand this function verifies if the resources have been already deleted, but its actual job is to clean them up, and the name should reflect that.
What something simple like `deleteResources`?

The same goes for the other similar functions here and in e2e test